srt-protocol/src/protocol/mod.rs

```diff
@@ -164,6 +169,12 @@ impl fmt::Debug for TimeSpan {
     }
 }
 
+impl From<Duration> for TimeSpan {
+    fn from(duration: Duration) -> TimeSpan {
+        TimeSpan::from_micros(duration.as_micros() as i32)
```

Comment:
Should this panic on overflow? Don't want to create weird situtations

Reply:
I was fine with it panicking, this really can’t ever overflow in practical use.

Reply:
I meant it should panic if there's an overflow, I'm not sure if `as i32` panics on overflow

Reply:
Yeah, an explicit panic would be good--want to avoid this situtation:

https://play.rust-lang.org/?version=stable&mode=debug&edition=2018&gist=e81f9381a6549cfbd5a00ff3f4eef5eb